### PR TITLE
Modifications to make Amrvis Spack compatible 

### PR DESCRIPTION
**Pull request by etpalmer63**
The three locations searched for `amrvis.defautls` are not compatible with a Spack installation. My solution is to accept a variable for the install path for `amrvis.defaults` at compile time. 

This is to address issue: https://github.com/AMReX-Codes/Amrvis/issues/13

### CODE REVIEW

File: GlobalUtilities.H [hunk 1]
```diff
@@ -8,6 +8,8 @@
 #include <AMReX_Box.H>
 #include <AMReX_Vector.H>
 
+#define CONFIG_FILE_PATH
```
[the remark this answers is not in the document]
Comment (etpalmer63) [reply GlobalUtilities.H]: Removed!

File: GlobalUtilities.cpp [hunk 2]
```diff
@@ -275,50 +275,63 @@ void AVGlobals::GetDefaults(const string &defaultsFile) {
...
 
+
   // try to find the defaultsFile
+
   string fullDefaultsFile;
 
-  fullDefaultsFile = "./" + defaultsFile;     // try dot first
+  const char *homePath = getenv("HOME");
+  int homePathLen = strlen(getenv("HOME"));
+  string homePathStr(homePath, homePathLen);
```
[the remark this answers is not in the document]
Comment (etpalmer63) [reply GlobalUtilities.cpp]: Yes it does. Thank you!

File: GNUmakefile [hunk 3]
```diff
@@ -180,6 +180,8 @@ endif
 #DEFINES += -DSCROLLBARERROR
 #DEFINES += -DFIXDENORMALS
 
+DEFINES += -DAMRVIS_CONFIG_FILEPATH='"${CONFIG_FILEPATH}"'
```
[the remark this answers is not in the document]
Comment (etpalmer63) [reply GNUmakefile]: Changed! :+1:

File: GlobalUtilities.cpp [hunk 4]
```diff
@@ -275,50 +275,63 @@ void AVGlobals::GetDefaults(const string &defaultsFile) {
...
+        break;
       }
+  }
+
+  if (!fileFound) {
+    if(ParallelDescriptor::IOProcessor()) {
+        cout << "Cannot find amrvis defaults file:  " << fullDefaultsFile << endl;
+        cout << "Using standard defaults." << endl;
     }
+
```
[the remark this answers is not in the document]
Comment (etpalmer63) [reply GlobalUtilities.cpp]: Yes. I put it back in inside the `(!fileFound)` logic.

File: GlobalUtilities.cpp [hunk 5]
```diff
@@ -275,50 +275,63 @@ void AVGlobals::GetDefaults(const string &defaultsFile) {
...
-    if(defs.fail()) {  // try ~/.  (hidden file)
-      if(ParallelDescriptor::IOProcessor()) {
-        cout << "Cannot find amrvis defaults file:  " << fullDefaultsFile << endl;
-      }
-      fullDefaultsFile  = getenv("HOME");
-      fullDefaultsFile += "/.";
-      fullDefaultsFile += defaultsFile;
+  bool fileFound = false;
+
+  for ( int i=0; i< fullDefaultsFileList.size(); ++i){
```
[the remark this answers is not in the document]
Comment (etpalmer63) [reply GlobalUtilities.cpp]: My C background is showing... I implemented the range-based loop and now I feel like I am writing python! :grin:

**Comment by etpalmer63** [comment]
Thanks for all the code suggestions!! This has been fun to figure out.

File: GlobalUtilities.cpp [hunk 6]
```diff
@@ -275,50 +275,59 @@ void AVGlobals::GetDefaults(const string &defaultsFile) {
...
-    if(defs.fail()) {  // try ~/.  (hidden file)
-      if(ParallelDescriptor::IOProcessor()) {
-        cout << "Cannot find amrvis defaults file:  " << fullDefaultsFile << endl;
-      }
-      fullDefaultsFile  = getenv("HOME");
-      fullDefaultsFile += "/.";
-      fullDefaultsFile += defaultsFile;
+  bool fileFound = false;
+
+  for (string fileLoc : fullDefaultsFileList){
```
Comment (WeiqunZhang) [inline GlobalUtilities.cpp:297]: Yes, it's influenced by python.  But string is not a simple type.  So this is better.
```suggestion
  for (string const& fileLoc : fullDefaultsFileList){
```

Comment (etpalmer63) [reply GlobalUtilities.cpp]: Good to know, made the change!

**Review by WeiqunZhang — APPROVED** [review]

**Pull request merged**